### PR TITLE
using hardhat as opposed to buidler

### PR DESCRIPTION
**Pull request by rajivpo**
Took at stab at this based on comment in #57 before it got self-assigned.

Needed to upgrade `@eth-optimism/smock` to `^0.2.1-alpha.0` to make use of `hardhat-storagelayout`

Tests pass, but not sure if there are additional considerations to ensure that this doesn't break anything and happy to implement any changes as requested.

### CODE REVIEW

File: cache/solidity-files-cache.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (tynes) [inline cache/solidity-files-cache.json:1]: Do you mind adding `cache/*` to the `.gitignore`?

Comment (smartcontracts) [reply cache/solidity-files-cache.json:1]: Will need to remove this folder from your git cache: `git rm --cached cache/solidity-files-cache.json`

**Comment by smartcontracts** [comment]
This is really great :-) thank you! Reviewing now.

**Review by smartcontracts — APPROVED** [review]

**Review by smartcontracts — APPROVED** [review]
Perfect. Let's get this merged!

**Pull request merged**